game/requesthandler.cc

```diff
@@ -1,376 +1,235 @@
-            if(songIdToDelete >= 0) {
-                gm->getCurrentPlayList().removeSong(songIdToDelete);
+#include "screen_playlist.hh"
+
+RequestHandler::RequestHandler(std::string, Songs &songs) : m_songs(songs) {
+	// register pages
+
+	// get all static pages from the www folder
+	m_server.set_base_dir(findFile("index.html").parent_path().string());
+
+	// GET API requests
+	m_server.Get("/api/language", [&](const httplib::Request&, httplib::Response& res) {
+		std::map<std::string, std::string> localeMap;
+		for (auto const &translationKey : GetTranslationKeys()) {
+		    localeMap[translationKey] = _(translationKey);
+		}
+		auto jsonRoot = nlohmann::json::object();
+		for (auto const &kv : localeMap) {
+		    std::string key = kv.first;
+		    //Hack to get an easy key value pair within the json object.
+		    if(key == "Web interface by Niek Nooijens and Arjan Speiard, for full credits regarding Performous see /docs/Authors.txt"){
+		        key = "Credits";
+		    }
+		    std::replace(key.begin(), key.end(), ' ','_');
+		    jsonRoot[UnicodeUtil::toLower(key)] = kv.second;
+		}
+		res.set_content(jsonRoot.dump(), "application/json");
+	});
+	m_server.Get("/api/getDataBase.json", [&](const httplib::Request &req, httplib::Response& res) {
+		m_songs.setFilter("");
+		bool sortDescending = false;
+		if (req.has_param("order") && req.get_param_value("order") == "descending") {
+			sortDescending = true;
+		}
+		const std::string sortItem = (req.has_param("sort")) ? req.get_param_value("sort") : "artist";
+		if(sortItem == "artist") {
+		    m_songs.sortSpecificChange(2, sortDescending);
+		} else if(sortItem == "title") {
+		    m_songs.sortSpecificChange(1, sortDescending);
+		} else if(sortItem == "language") {
+		    m_songs.sortSpecificChange(6, sortDescending);
+		} else if(sortItem == "edition") {
+		    m_songs.sortSpecificChange(3, sortDescending);
+		}
+		auto jsonRoot = nlohmann::json::array();
+		for (size_t i=0; i< m_songs.size(); i++) {
+			jsonRoot.push_back({
+				{"Title", m_songs[i]->title},
+				{"Artist", m_songs[i]->artist},
+				{"Edition", m_songs[i]->edition},
+				{"Language", m_songs[i]->language},
+				{"Creator", m_songs[i]->creator},
+				{"name", m_songs[i]->artist + " " + m_songs[i]->title},
+			});
+		}
+		res.set_content(jsonRoot.dump(), "application/json");
+	});
+	m_server.Get("/api/getCurrentPlaylist.json", [&](const httplib::Request &, httplib::Response& res) {
+		Game* gm = Game::getSingletonPtr();
+		auto jsonRoot = nlohmann::json::array();
+		for (auto const& song : gm->getCurrentPlayList().getList()) {
+			jsonRoot.push_back({
+				{"Title", song->title},
+				{"Artist", song->artist},
+				{"Edition", song->edition},
+				{"Language", song->language},
+				{"Creator", song->creator},
+				{"name", song->artist + " " + song->title},
+				{"Duration", song->getDurationSeconds()},
+			});
+		}
+		res.set_content(jsonRoot.dump(), "application/json");
```

Comment:
If no status is set it defaults to `200 OK`? I think it'd be nice to always set it explicitly (for each api endpoint)